aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -325,6 +325,12 @@ Tensor& cat_out_cuda(Tensor& out, TensorList inputs, int64_t dimension) {
   TORCH_CHECK(inputs.size() > 0, "invalid number of inputs ", inputs.size());
   TORCH_CHECK(dimension >= 0, "invalid dimension ", dimension);
 
+  for (const Tensor& t: inputs) {
+    TORCH_CHECK(t.device() == notSkippedTensor->device(),
```

Comment:
@gchanan but the check is performed for every input tensor, whether it was skipped or not. `notSkippedTensor` is simply an arbitrary tensor to compare all the others against.

It might be indeed somewhat confusing. `inputs[0]` would be just as good but I used `notSkippedTensor` since it's used for the same purpose everywhere else in this function.

Reply:
yes, good point.  I think `notSkippedTensor` is correctly to use as `inputs[0]` could be skipped.